backend/controller/role.go

```diff
@@ -41,6 +41,14 @@ func CreateRole(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		http.Error(w, "Malformed request", http.StatusBadRequest)
 	}
+	doors := role.Doors
+	for i, _ := range doors {
+		res := db.DB.First(&doors[i])
+		if res.Error != nil {
+			http.Error(w, "One or more invalid doors entered", http.StatusBadRequest)
+			return
+		}
+	}
```

Comment:
This seems wildly inefficient, can we do something like this and compare the lengths:

![Screenshot 2024-03-11 at 3 38 30 PM](https://github.com/lynx-locks/lynx-web/assets/55325093/f1b7b7f2-142e-4240-984b-18da011e8329)
![Screenshot 2024-03-11 at 3 39 10 PM](https://github.com/lynx-locks/lynx-web/assets/55325093/5f5aeb10-af3b-4d97-8c1e-11d5218e359e)
![Screenshot 2024-03-11 at 3 39 01 PM](https://github.com/lynx-locks/lynx-web/assets/55325093/fe41a17d-d143-476a-96dc-1bcd1314d2a5)
![Screenshot 2024-03-11 at 3 38 50 PM](https://github.com/lynx-locks/lynx-web/assets/55325093/16f73add-2a72-4c7e-a47d-cfecd865104c)

Reference: https://gorm.io/docs/query.html#Retrieving-all-objects